src/circuit.rs

```diff
@@ -438,18 +436,30 @@ impl plonk::Circuit<pallas::Base> for Circuit {
                 (magnitude, sign)
             };
 
+            let v_net = ScalarFixedShort::new(
+                ecc_chip.clone(),
+                layouter.namespace(|| "v_net"),
+                v_net_magnitude_sign.clone(),
+            )?;
+            let rcv = ScalarFixed::new(
+                ecc_chip.clone(),
+                layouter.namespace(|| "rcv"),
+                self.rcv.as_ref().map(|rcv| rcv.inner()),
+            )?;
+
             let cv_net = gadget::value_commit_orchard(
                 layouter.namespace(|| "cv_net = ValueCommit^Orchard_rcv(v_net)"),
                 ecc_chip.clone(),
-                v_net.clone(),
-                self.rcv.as_ref().map(|rcv| rcv.inner()),
+                v_net,
+                rcv,
             )?;
 
             // Constrain cv_net to equal public input
             layouter.constrain_instance(cv_net.inner().x().cell(), config.primary, CV_NET_X)?;
             layouter.constrain_instance(cv_net.inner().y().cell(), config.primary, CV_NET_Y)?;
 
-            v_net
+            // Return the magnitude and sign so we can use them in the Orchard gate.
```

Comment:
nonblocking: What's the "Orchard gate"

Reply:
The gate controlled by `q_orchard`, defined earlier.